client/reader/controller.js

```diff
@@ -213,9 +241,34 @@ module.exports = {
 		);
 	},
 
+	feedDiscovery: function( context, next ) {
+		var FeedStore = require( 'lib/feed-store' ),
+			FeedStoreActions = require( 'lib/feed-store/actions' ),
+			FeedError = require( 'reader/feed-error' );
+
+		if ( ! context.params.feed_id.match( /^\d+$/ ) ) {
+			FeedStoreActions.discover( context.params.feed_id );
```

Comment:
Indeed. I've implemented it this way and it really made things much simpler and less fragile.